diff -uNrp rtslib-3.0-pre4.orig/rtslib/config_parser.py rtslib-3.0-pre4/rtslib/config_parser.py
--- rtslib-3.0-pre4.orig/rtslib/config_parser.py	2014-10-30 19:03:58.000000000 +0100
+++ rtslib-3.0-pre4/rtslib/config_parser.py	2017-03-19 10:32:22.535398455 +0100
@@ -225,7 +225,7 @@ class PolicyParser(ConfigParser):
         value = tokin[0].asDict()
         ref_path = value.get('ref_path')
         if ref_path is not None:
-            ref_path = " ".join(ref_path.asList())
+            ref_path = " ".join(ref_path)
         tokout = {'type': 'attr',
                   'line': pp.lineno(idx, source),
                   'col': pp.col(idx, source),
