From 84cf8e4dde217efb8de2468b6f88e59dfffce2bb Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 16 Jun 2014 15:04:07 +0200
Subject: [PATCH] 279 inline correctly

---
 vmmon-only/linux/driver.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vmmon-only/linux/driver.c b/vmmon-only/linux/driver.c
index 6e6f088..6f46786 100644
--- a/vmmon-only/linux/driver.c
+++ b/vmmon-only/linux/driver.c
@@ -1332,7 +1332,7 @@ LinuxDriverReadTSC(void *data,   // OUT: TSC values
  *-----------------------------------------------------------------------------
  */
 
-__attribute__((always_inline)) static Bool
+INLINE_SINGLE_CALLER static Bool
 LinuxDriverSyncReadTSCs(uint64 *delta) // OUT: TSC max - TSC min
 {
    TSCDelta tscDelta;
-- 
1.8.5.5

