From c05fdac3df5e68eb04da953160577ffea60d2b25 Mon Sep 17 00:00:00 2001
From: Root <root@cera.cz>
Date: Mon, 14 Nov 2016 15:02:03 +0100
Subject: [PATCH 3/3] fix PCI ID warning

---
 vmnet-only/netif.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/vmnet-only/netif.c b/vmnet-only/netif.c
index 5ac4e69..ed470ba 100644
--- a/vmnet-only/netif.c
+++ b/vmnet-only/netif.c
@@ -38,6 +38,7 @@
 
 #include <linux/proc_fs.h>
 #include <linux/file.h>
+#include <linux/pci_ids.h>
 
 #include "vnetInt.h"
 #include "compat_netdevice.h"
-- 
2.7.3

