diff --git a/benchmark/CMakeLists.txt b/benchmark/CMakeLists.txt
index d96a2e0..fe18f4c 100644
--- a/benchmark/CMakeLists.txt
+++ b/benchmark/CMakeLists.txt
@@ -15,7 +15,7 @@ set (benchmark_sources
 ADD_LIBRARY(benchmarks SHARED ${ADDMATHFUNC} ${benchmark_sources})
 
 #TARGET_LINK_LIBRARIES(benchmarks cl m ${OPENGL_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
-TARGET_LINK_LIBRARIES(benchmarks cl m)
+TARGET_LINK_LIBRARIES(benchmarks OpenCL m)
 
 ADD_EXECUTABLE(benchmark_run benchmark_run.cpp)
 TARGET_LINK_LIBRARIES(benchmark_run benchmarks)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9b41932..e5f5259 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -110,9 +110,10 @@ endif (OCLIcd_FOUND)
 SET(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-Bsymbolic,--allow-shlib-undefined")
 
 link_directories (${LLVM_LIBRARY_DIR} ${DRM_LIBDIR})
-add_library(cl SHARED ${OPENCL_SRC})
+add_library(OpenCL SHARED ${OPENCL_SRC})
+set_target_properties(OpenCL PROPERTIES VERSION 1.0.0 SOVERSION 1)
 target_link_libraries(
-                      cl
+                      OpenCL
                       ${X11_LIBRARIES}
                       ${XEXT_LIBRARIES}
                       ${XFIXES_LIBRARIES}
@@ -122,4 +123,4 @@ target_link_libraries(
                       ${CMAKE_DL_LIBS}
                       ${OPENGL_LIBRARIES}
                       ${OPTIONAL_EGL_LIBRARY})
-install (TARGETS cl LIBRARY DESTINATION ${BEIGNET_INSTALL_DIR})
+install (TARGETS OpenCL LIBRARY DESTINATION ${BEIGNET_INSTALL_DIR})
diff --git a/utests/CMakeLists.txt b/utests/CMakeLists.txt
index d786a45..075abf9 100644
--- a/utests/CMakeLists.txt
+++ b/utests/CMakeLists.txt
@@ -218,7 +218,7 @@ endif()
 
 ADD_LIBRARY(utests SHARED ${ADDMATHFUNC} ${utests_sources})
 
-TARGET_LINK_LIBRARIES(utests cl m ${OPENGL_LIBRARIES} ${UTESTS_REQUIRED_EGL_LIB} ${CMAKE_THREAD_LIBS_INIT})
+TARGET_LINK_LIBRARIES(utests OpenCL m ${OPENGL_LIBRARIES} ${UTESTS_REQUIRED_EGL_LIB} ${CMAKE_THREAD_LIBS_INIT})
 
 ADD_EXECUTABLE(utest_run utest_run.cpp)
 TARGET_LINK_LIBRARIES(utest_run utests)
