diff --git a/lib/irrlicht/source/Irrlicht/CIrrDeviceLinux.cpp b/lib/irrlicht/source/Irrlicht/CIrrDeviceLinux.cpp
index 5ee1d36..9269b64 100644
--- a/lib/irrlicht/source/Irrlicht/CIrrDeviceLinux.cpp
+++ b/lib/irrlicht/source/Irrlicht/CIrrDeviceLinux.cpp
@@ -912,6 +912,13 @@ bool CIrrDeviceLinux::createWindow()
 				&visTempl, &visNumber);
 			visTempl.depth -= 8;
 		}
+
+		if (!visual && !CreationParams.WithAlphaChannel)
+		{
+			visTempl.depth = 32;
+			visual = XGetVisualInfo(display, VisualScreenMask|VisualDepthMask,
+									&visTempl, &visNumber);
+		}
 	}
 
 	if (!visual)
