diff -urpN gst-libav-1.2.4.orig/ext/libav/gstavauddec.c gst-libav-1.2.4/ext/libav/gstavauddec.c
--- gst-libav-1.2.4.orig/ext/libav/gstavauddec.c	2014-01-15 02:19:58.000000000 -0800
+++ gst-libav-1.2.4/ext/libav/gstavauddec.c	2014-06-26 21:35:45.822494686 -0700
@@ -464,7 +464,7 @@ gst_ffmpegauddec_audio_frame (GstFFMpegA
     GstBuffer ** outbuf, GstFlowReturn * ret)
 {
   gint len = -1;
-  gint have_data = AVCODEC_MAX_AUDIO_FRAME_SIZE;
+  gint have_data = 0;
   AVPacket packet;
   AVFrame frame;
 
@@ -478,7 +478,7 @@ gst_ffmpegauddec_audio_frame (GstFFMpegA
   GST_DEBUG_OBJECT (ffmpegdec,
       "Decode audio: len=%d, have_data=%d", len, have_data);
 
-  if (len >= 0 && have_data > 0) {
+  if (len >= 0 && have_data) {
     BufferInfo *buffer_info = frame.opaque;
     gint nsamples, channels, byte_per_sample;
     gsize output_size;
@@ -572,7 +572,8 @@ gst_ffmpegauddec_audio_frame (GstFFMpegA
       gst_buffer_fill (*outbuf, 0, frame.data[0], output_size);
     }
 
-    GST_DEBUG_OBJECT (ffmpegdec, "Buffer created. Size: %d", have_data);
+    GST_DEBUG_OBJECT (ffmpegdec, "Buffer created. Size: %" G_GSIZE_FORMAT,
+        output_size);
 
     /* Reorder channels to the GStreamer channel order */
     if (ffmpegdec->needs_reorder) {
