diff -uNrp xtables-addons-2.2.orig/extensions/xt_condition.c xtables-addons-2.2/extensions/xt_condition.c
--- xtables-addons-2.2.orig/extensions/xt_condition.c	2013-03-31 21:14:37.000000000 +0200
+++ xtables-addons-2.2/extensions/xt_condition.c	2013-05-03 09:37:07.320547908 +0200
@@ -155,8 +155,8 @@ static int condition_mt_check(const stru
 	var->status_proc->read_proc  = condition_proc_read;
 	var->status_proc->write_proc = condition_proc_write;
 	list_add(&var->list, &conditions_list);
-	var->status_proc->uid = condition_uid_perms;
-	var->status_proc->gid = condition_gid_perms;
+	var->status_proc->uid = make_kuid(&init_user_ns, condition_uid_perms);
+	var->status_proc->gid = make_kgid(&init_user_ns, condition_gid_perms);
 	mutex_unlock(&proc_lock);
 	info->condvar = var;
 	return 0;
diff -uNrp xtables-addons-2.2.orig/extensions/xt_DNETMAP.c xtables-addons-2.2/extensions/xt_DNETMAP.c
--- xtables-addons-2.2.orig/extensions/xt_DNETMAP.c	2013-03-31 21:14:37.000000000 +0200
+++ xtables-addons-2.2/extensions/xt_DNETMAP.c	2013-05-03 09:36:55.635397868 +0200
@@ -41,8 +41,8 @@ MODULE_ALIAS("ipt_DNETMAP");
 
 static unsigned int default_ttl = 600;
 static unsigned int proc_perms = S_IRUGO | S_IWUSR;
-static unsigned int proc_uid;
-static unsigned int proc_gid;
+static kuid_t proc_uid = GLOBAL_ROOT_UID;
+static kgid_t proc_gid = GLOBAL_ROOT_GID;
 static unsigned int default_hash_size = 256;
 static unsigned int hash_size = 256;
 static unsigned int disable_log;
diff -uNrp xtables-addons-2.2.orig/extensions/xt_quota2.c xtables-addons-2.2/extensions/xt_quota2.c
--- xtables-addons-2.2.orig/extensions/xt_quota2.c	2013-03-31 21:14:37.000000000 +0200
+++ xtables-addons-2.2/extensions/xt_quota2.c	2013-05-03 09:37:01.405471851 +0200
@@ -131,8 +131,8 @@ q2_get_counter(const struct xt_quota_mti
 	p->data         = e;
 	p->read_proc    = quota_proc_read;
 	p->write_proc   = quota_proc_write;
-	p->uid          = quota_list_uid;
-	p->gid          = quota_list_gid;
+	p->uid          = make_kuid(&init_user_ns, quota_list_uid);
+	p->gid          = make_kgid(&init_user_ns, quota_list_gid);
 	list_add_tail(&e->list, &counter_list);
 	spin_unlock_bh(&counter_list_lock);
 	return e;
