diff -uNrp krb5-auth-dialog-0.17.orig/src/ka-applet.c krb5-auth-dialog-0.17/src/ka-applet.c
--- krb5-auth-dialog-0.17.orig/src/ka-applet.c	2010-07-26 13:23:17.000000000 +0200
+++ krb5-auth-dialog-0.17/src/ka-applet.c	2011-04-01 11:24:27.382518360 +0200
@@ -488,11 +488,15 @@ ka_send_event_notification (KaApplet *ap
 
 	notify_icon = icon ? icon : "krb-valid-ticket";
 
-	applet->priv->notification = \
+	applet->priv->notification =
+#if ((NOTIFY_VERSION_MAJOR == 0) && (NOTIFY_VERSION_MINOR < 7))
 		notify_notification_new_with_status_icon(summary,
 							 message,
 							 notify_icon,
 							 applet->priv->tray_icon);
+#else
+		notify_notification_new(summary, message, notify_icon);
+#endif
 
 	notify_notification_set_urgency (applet->priv->notification, NOTIFY_URGENCY_NORMAL);
 	notify_notification_add_action (applet->priv->notification,
