diff -uNrp xen-4.1.1.orig/xen/arch/x86/traps.c xen-4.1.1/xen/arch/x86/traps.c
--- xen-4.1.1.orig/xen/arch/x86/traps.c	2011-06-14 18:03:45.000000000 +0200
+++ xen-4.1.1/xen/arch/x86/traps.c	2011-09-13 13:34:08.883500036 +0200
@@ -2371,6 +2371,15 @@ static int emulate_privileged_op(struct
             if ( wrmsr_safe(regs->ecx, msr_content) != 0 )
                 goto fail;
             break;
+        case MSR_IA32_TURBO_POWER_CURRENT_LIMIT:
+        case MSR_IA32_TURBO_RATIO_LIMIT:
+            if ( boot_cpu_data.x86_vendor != X86_VENDOR_INTEL )
+                goto fail;
+            if ( !IS_PRIV(v->domain) )
+                break;
+            if ( wrmsr_safe(regs->ecx, msr_content) != 0 )
+                goto fail;
+            break;
         default:
             if ( wrmsr_hypervisor_regs(regs->ecx, msr_content) )
                 break;
diff -uNrp xen-4.1.1.orig/xen/include/asm-x86/msr-index.h xen-4.1.1/xen/include/asm-x86/msr-index.h
--- xen-4.1.1.orig/xen/include/asm-x86/msr-index.h	2011-06-14 18:03:46.000000000 +0200
+++ xen-4.1.1/xen/include/asm-x86/msr-index.h	2011-09-13 13:35:56.505116789 +0200
@@ -332,6 +332,9 @@
 #define MSR_IA32_TSC_DEADLINE		0x000006E0
 #define MSR_IA32_ENERGY_PERF_BIAS	0x000001b0
 
+#define MSR_IA32_TURBO_POWER_CURRENT_LIMIT	0x000001ac
+#define MSR_IA32_TURBO_RATIO_LIMIT	0x000001ad
+
 /* Intel Model 6 */
 #define MSR_P6_EVNTSEL0			0x00000186
 #define MSR_P6_EVNTSEL1			0x00000187
